geetools/ee_dictionary.py

```diff
@@ -80,3 +82,172 @@ def getMany(self, list: list | ee.List) -> ee.List:
+                    'ADM0_CODE': 13
+                    'Shape_Area': 3.13783139285
+                  },
+                }
+            ]}
+
+        3. ee.List: when values are a ee.List of numbers or strings, the
+        keys will be saved in the column "system:index" and the values in
+        as many columns as items in the list. The column name pattern is
+        "value_{i}" where i is the position of the element in the list.
+
+        .. code-block:: python
+
+            import ee, geetools
+
+            ee.Initialize()
+
+            d = ee.Dictionary({
+              "Argentina": [12, 278.289196625],
+              "Armenia": [13, 3.13783139285],
+            })
+            d.geetools.toTable().getInfo()
+
+            >> {
+              'type': FeatureCollection,
+              'columns': {
+                'system:index': String,
+                'value_0': Integer,
+                'value_1': Float
+              },
+              features: [
+                {
+                  'type': Feature,
+                  'geometry': null,
+                  'id': 'Argentina',
+                  'properties': {
+                    'value_0': 12
+                    'value_1': 278.289196625
+                  },
+                },
+                {
+                  'type': Feature,
+                  'geometry': null,
+                  'id': 'Armenia',
+                  'properties': {
+                    'value_0': 13
+                    'value_1': 3.13783139285
+                  },
+                }
+            ]}
+
+        Parameters:
+            valueType: this will define how to process the values. In case of
+            ee.List the values will be stored in columns named `value_<position>`.
+            In case of a ee.Dictionary, column names will be created from the keys.
+            For any other type, it will return a table with one feature with
+            one column per key.
+
+        Returns:
+            a ee.FeatureCollection in which the keys of the ee.Dictionary are
+            in the `system:index` and the values are in new columns.
+        """
+
+        def features_from_dict(key, value) -> ee.Feature:
+            index = {"system:index": ee.String(key)}
+            props = ee.Dictionary(value).combine(index)
+            return ee.Feature(None, props)
+
+        def features_from_list(key, value) -> ee.Feature:
+            index = {"system:index": ee.String(key)}
+            values = ee.List(value)
+            columns = ee.List.sequence(0, values.size().subtract(1))
+            columns = columns.map(lambda k: ee.String("value_").cat(ee.Number(k).toInt()))
+            props = ee.Dictionary.fromLists(columns, values).combine(index)
+            return ee.Feature(None, props)
+
+        def features_from_any(key, value) -> ee.Feature:
+            props = {"system:index": ee.String(key), "value": value}
+            return ee.Feature(None, props)
+
+        if valueType == ee.Dictionary:
```

Comment:
insterad of object could we use a lighter typing like str with 3 options: "dictionary", "list", "scalar".
In this case scalar would be to mimick pandas terminology but that could of course be anything else. 

If we go this way then instead of an if you could create a dictionary like: 
```python 
make_features{
    "list": feature_from_list,
    "dictionary": feature_from_dictionary,
    "scalar": feature_from_any
}
features = self._obj.map(make_features).values()

return ee.FeatureCollection(features)
```

That's the most pythonic way of doing a switch in python. Alternatively as we are only supporting Python >=3.10, you could also use the `match` statement which have the "else" option to raise error.

Reply:
Totally agree. I also prefer this way and have done it many times. Not sure why this time I used the old fashion `if.. else..` statement, but I can improve it for sure.

Reply:
I'm not convinced about `scalar` 🤔 What about `any`? we could let the user have any type of value and it will fail if it's not suitable as a valid property value.